# HG changeset patch
# Parent cccdcb177bb694ffa77966e0499ddacba722fd69
Allow XUL elements to use Input Method Editing (IME). Komodo uses this on it's
xul:scintilla elements to allow IME event handling, which it then passes down
to the SciMoz plugin component.

diff --git a/content/events/src/nsIMEStateManager.cpp b/content/events/src/nsIMEStateManager.cpp
--- a/content/events/src/nsIMEStateManager.cpp
+++ b/content/events/src/nsIMEStateManager.cpp
@@ -234,16 +234,23 @@ nsIMEStateManager::GetNewIMEState(nsPres
     // Even if there are no focused content, the focused document might be
     // editable, such case is design mode.
     nsIDocument* doc = aPresContext->Document();
     if (doc && doc->HasFlag(NODE_IS_EDITABLE))
       return nsIContent::IME_STATUS_ENABLE;
     return nsIContent::IME_STATUS_DISABLE;
   }
 
+  if (aContent->AttrValueIs(kNameSpaceID_None,
+                            nsGkAtoms::allowIME,
+                            NS_LITERAL_STRING("true"),
+                            eCaseMatters)) {
+    return nsIContent::IME_STATUS_ENABLE;
+  }
+
   return aContent->GetDesiredIMEState();
 }
 
 // Helper class, used for IME enabled state change notification
 class IMEEnabledStateChangedEvent : public nsRunnable {
 public:
   IMEEnabledStateChangedEvent(PRUint32 aState)
     : mState(aState)
